app/TerminalViewController.m

```diff
@@ -325,16 +325,23 @@ - (void)keyboardDidSomething:(NSNotification *)notification {
         return;
 
     CGRect keyboardFrame = [notification.userInfo[UIKeyboardFrameEndUserInfoKey] CGRectValue];
+    keyboardFrame = [self.view convertRect:keyboardFrame fromView:self.view.window];
     if (CGRectEqualToRect(keyboardFrame, CGRectZero))
         return;
     NSLog(@"%@ %@", notification.name, [NSValue valueWithCGRect:keyboardFrame]);
     self.hasExternalKeyboard = keyboardFrame.size.height < 100;
-    CGFloat pad = UIScreen.mainScreen.bounds.size.height - keyboardFrame.origin.y;
-    if (pad != keyboardFrame.size.height) {
-        pad = 0; // keyboard is not right at the bottom of the screen, must be floating or something
-    }
-    if (pad == 0) {
-        pad = self.view.safeAreaInsets.bottom;
+    CGFloat pad = self.view.bounds.size.height - keyboardFrame.origin.y;
```

Comment:
Honestly the only way I can really review this is ask what ios versions you tested it on

Reply:
iOS 12-16, and I got @Mnpn to do iOS 11